### PR TITLE
Added Food101 Dataset

### PR DESCRIPTION
**Pull request by SoumyadeepJana**
@Conchylicultor  Kindly review :)
Dataset Request : https://github.com/tensorflow/datasets/issues/312

### CODE REVIEW

**Comment by SoumyadeepJana** [comment]
> The test data should not contain any copyrighted material.

It doesn't I think, All those images are not copyrighted

File: tensorflow_datasets/image/food101.py [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+            description=(_DESCRIPTION),
+            
+            features=tfds.features.FeaturesDict({
+                "image":tfds.features.Image(),
+                "label": tfds.features.ClassLabel(names_file=names_file),
+            }),
+
+            supervised_keys=("image", "label"),
+            
+            urls=[_BASE_URL],
```
Comment (us) [inline tensorflow_datasets/image/food101.py]: you should add dataset [link](https://www.vision.ee.ethz.ch/datasets_extra/food-101/) not a download link.

File: tensorflow_datasets/image/food101.py [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+            citation=_CITATION
+        )
+
+  def _split_generators(self, dl_manager):
+        """Define Splits"""
+        path = dl_manager.download_and_extract(_BASE_URL)
+
+        return [
+            tfds.core.SplitGenerator(
+                name="train",
```
Comment (cyfra) [inline tensorflow_datasets/image/food101.py]: nit: please use 

name=tfds.Split.TRAIN

Comment (SoumyadeepJana) [reply tensorflow_datasets/image/food101.py]: Implemented the changes.Kindly review. :)

Comment (SoumyadeepJana) [reply tensorflow_datasets/image/food101.py]: Is there any specific reason for using tfds.Split.TRAIN??

**Pull request merged**